src/chainstate/stacks/db/mod.rs

```diff
@@ -908,10 +908,9 @@ impl StacksChainState {
     /// Get the appropriate MARF index hash to use to identify a chain tip, given a parent block
     /// header
     pub fn get_index_hash(burn_hash: &BurnchainHeaderHash, header: &StacksBlockHeader) -> BlockHeaderHash {
-        if header.is_genesis() {
+        if burn_hash == &FIRST_BURNCHAIN_BLOCK_HASH {
```

Comment:
Do all the tests pass with this change?

Reply:
No -- the miner tests fail, since calculate_miner_rewards breaks in those tests. What do you think the behavior here should be?


Previously, the block construction worked something like this:

genesis execution: written to make_index_block_hash([0; 32], [0; 32]), parent is make_index_block_hash([ff; 32], [ff; 32]), which is not the sentinel.
1st mined block: parent is [0; 32], so this function replaced parent with the sentinel.

This change made it so that the genesis execution's parent is the sentinel, and the first mined block's parent is the genesis execution.

Reply:
Having tested the miner code, this change is fine, as long as you remove the `assert!` from `calculate_miner_rewards`.